lib/Net/Async/Github.pm

```diff
@@ -808,8 +843,8 @@ sub http {
                 pipeline                 => 1,
                 max_in_flight            => 4,
                 decode_content           => 1,
-                timeout                  => $self->timeout,
                 user_agent               => 'Mozilla/4.0 (perl; Net::Async::Github; TEAM@cpan.org)',
+                $self->timeout ? (timeout => $self->timeout) : (),
             )
```

Comment:
Do we need this check $self->timeout ? here since $self->timeout subroutine always return either assigned or default one

Reply:
Please consider the case that `timeout` is `0`. Then here it will be false, and then no timeout argument . Then Net::Async::HTTP will never `timeout`